Celeste.Mod.mm/Patches/Level.cs

```diff
@@ -709,31 +1524,85 @@ static partial class MonoModRules {
+            cursor.Emit(OpCodes.Ldarg_0); // level
+            Instruction newFinallyEnd = cursor.Prev;
+            cursor.Emit(OpCodes.Ldnull); // level, null
+            cursor.Emit(OpCodes.Dup);    // level, null, null
+            cursor.Emit(OpCodes.Ldc_I4_0); // level, null, null, false
+            cursor.Emit(OpCodes.Call, m_Level_SetTemporaryEntityData); // level.SetTemporaryEntityData(null, null, false);
+            // fix end of finally block
+            foreach (ExceptionHandler handler in context.Body.ExceptionHandlers.Where(handler => handler.HandlerEnd == oldFinallyEnd)) {
+                handler.HandlerEnd = newFinallyEnd;
+                break;
+            }
+            // Code change:
+            //   foreach (EntityData trigger in levelData.Triggers) {
+            // +    level.SetTemporaryEntityData(entity3, levelData, isTrigger: true);  This will always be changed at the start of each loop, so i just need to set it to null after the loop.
+            //      int entityID2 = trigger.ID;
+            //      EntityID entityID3 = new EntityID(levelData.Name, entityID2);       Kept for mod parity
+            // +    entityID = trigger.EntityID;                                        ID offset is managed in LevelData.CreateEntityData and verified with FixEntityData in SetTemporaryEntityData
+            //      ..., switch(...) { cases ... }
+            //   }
+            // + level.SetTemporaryEntityData(null, null, false);                       ince we don't null at the end of the switch statement, we need to null here. Requires HandlerException resolution
+            //   foreach (DecalData fgDecal in levelData.FgDecals) ...
+            cursor.GotoNext(MoveType.AfterLabel, instr => instr.MatchLdloc(46)); // trigger
+            cursor.Emit(OpCodes.Ldarg_0);
+            cursor.Emit(OpCodes.Ldloc, 46); // trigger
+            cursor.Emit(OpCodes.Ldloc, 3); // levelData
+            cursor.Emit(OpCodes.Ldc_I4_1); // `true`
+            cursor.Emit(OpCodes.Call, m_Level_SetTemporaryEntityData); // level.SetTemporaryEntityData(trigger, levelData, isTrigger: true);
+            cursor.GotoNext(MoveType.Before, instr => instr.MatchLdarg(0));
+            cursor.Emit(OpCodes.Ldloc, 46); // trigger
+            cursor.Emit(OpCodes.Ldfld, f_EntityData_EntityID); // trigger.EntityID
+            cursor.Emit(OpCodes.Stloc, 48); // entityID = trigger.EntityID;
+
             ILLabel continueLabel = null;
             cursor.GotoNext(MoveType.After, instr => instr.MatchBrtrue(out continueLabel));
             // add
             // || _IsInDoNotLoadIncreased(levelData, trigger)
             // to if condition for continue to handle triggers that already add 10000000 to their DoNotLoad entry
             cursor.EmitLdarg0();
-            cursor.EmitLdloc(v_levelData);
-            cursor.EmitLdloc(v_entityData);
+            cursor.EmitLdloc(3); // levelData
```

Comment:
Why are you making the patch less readable by replacing variable references with magic numbers?